tla-io/src/main/scala/at/forsyte/apalache/io/OutputManager.scala

```diff
@@ -150,4 +151,68 @@ object OutputManager {
       val rundir = Files.createTempDirectory(outDir, s"${specName}_${nicedate}T${nicetime}_")
       runDirOpt = Some(rundir)
     }
+
+  /** Create a PrintWriter to the file formed by appending `fileParts` to the `base` file */
+  def printWriter(base: File, fileParts: String*): PrintWriter = {
+    val file = fileParts.foldLeft(base)((file, part) => new File(file, part))
+    new PrintWriter(new FileWriter(file))
+  }
+
+  /** Create a PrintWriter to the file formed by appending `fileParts` to the `base` file */
+  def printWriter(base: Path, fileParts: String*): PrintWriter = {
+    printWriter(base.toFile(), fileParts: _*)
+  }
+
+  /**
+   *  Create a PrintWriter to the file formed by appending `fileParts` to the `base` file
+   *
+   * E.g., to create a writer to the file `foo/bar/bas.json`:
+   *
+   *    val w = printWriter("foo", "bar", "baz.json")
+   */
+  def printWriter(base: String, fileParts: String*): PrintWriter = {
+    printWriter(Paths.get(base), fileParts: _*)
+  }
+
+  /** Apply f to the writer w, being sure to close w */
+  private def withWriter(f: PrintWriter => Unit)(w: PrintWriter) = {
+    try {
+      f(w)
+    } finally {
+      w.close()
+    }
+  }
+
+  def withWriterToFile(file: File)(f: PrintWriter => Unit): Unit = {
+    withWriter(f)(printWriter(file))
+  }
+
+  /** Applies `f` to a PrintWriter created by appending the `parts` to the `runDir` */
+  def withWriterRelativeToRunDir(parts: String*)(f: PrintWriter => Unit): Unit = {
+    val w = printWriter(runDirOpt.get, parts: _*)
+    withWriter(f)(w)
+  }
+
+  /**
+   * Conditionally applies a function to a PrintWriter constructed relative to the intermediate directory
+   *
+   * @param parts path parts describing a path relative to the intermediate directory (all parents must exist)
+   * @param f a function that will be applied to the the PrintWriter, if the `IntermediateFlag` is set.
+   * @returns `true` if the `IntermediateFlag` is set the output is written, otherwise `false`
+   * If the IntermediateFlag is true, then this applies `f` to the PrintWriter
+   * created by appending the `parts` to the intermediate output dir, and returns `true`. Otherwise, it is false
+   */
+  def withWriterRelativeToIntermediateDir(parts: String*)(f: PrintWriter => Unit): Boolean = {
+    if (flags(Names.IntermediateFlag)) {
+      val intermediateDir = new File(runDirOpt.get.toFile(), Names.IntermediateFoldername)
+      if (!intermediateDir.exists()) {
+        intermediateDir.mkdir()
```

Comment:
`mkdir` may return `false`, if it is unable to create a directory. Do we have to do anything about that? I guess, the line below will fail with an IO exception in this case, which may be sufficient.

Reply:
Good call. Would be nicer to give a nice user error in this case.